core/executor/base_executor.go

```diff
@@ -15,9 +17,32 @@ type BaseExecutor struct {
 	bc      IBlockchain
 	tx      *tx.Tx
 	iTxInfo legendTxTypes.TxInfo
+
+	// Affected states.
+	dirtyAccountsAndAssetsMap map[int64]map[int64]bool
+	dirtyLiquidityMap         map[int64]bool
+	dirtyNftMap               map[int64]bool
+}
+
+func NewBaseExecutor(bc IBlockchain, tx *tx.Tx, txInfo legendTxTypes.TxInfo) BaseExecutor {
+	return BaseExecutor{
+		bc:      bc,
+		tx:      tx,
+		iTxInfo: txInfo,
+
+		dirtyAccountsAndAssetsMap: make(map[int64]map[int64]bool, 0),
+		dirtyLiquidityMap:         make(map[int64]bool, 0),
+		dirtyNftMap:               make(map[int64]bool, 0),
+	}
 }
 
 func (e *BaseExecutor) Prepare() error {
+	err := e.bc.StateDB().PrepareAccountsAndAssets(e.dirtyAccountsAndAssetsMap)
+	if err != nil {
+		logx.Errorf("prepare accounts and assets failed: %s", err.Error())
+		return errors.New("internal error")
```

Comment:
I will suggest return the err rather than `internal error`

Reply:
This is reused by api server, need to format the error return to the user.